src/main/java/medic/gateway/alert/PromptToSetAsDefaultMessageAppActivity.java

```diff
@@ -25,52 +26,80 @@ public PromptToSetAsDefaultMessageAppActivity() {
 	}
 
 //> EVENT HANDLERS
+
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
-		log("Starting...");
-
-		String currentDefaultSmsApp = android.provider.Settings.Secure.getString(getContentResolver(),
-				// this string is copied from android.provider.Settings.Secure.SMS_DEFAULT_APPLICATION
-				// because directly referencing this constant throws a compile error, presumably
-				// because of legacy support.
-				"sms_default_application");
-		log("Current default SMS app is %s", currentDefaultSmsApp);
+		log("Starting view for PromptToSetAsDefaultMessageAppActivity...");
 
 		setContentView(R.layout.set_as_default_messaging_app);
-
 		String appName = getAppName(this);
 		setText(this, R.id.txtDefaultMessageAppWarning, R.string.txtDefaultMessageAppWarning, appName);
 		setText(this, R.id.txtDefaultMessageAppPrompt, R.string.txtDefaultMessageAppPrompt, appName);
 	}
 
 	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-		switch(requestCode) {
+		switch (requestCode) {
 			case REQUEST_CHANGE_DEFAULT_MESSAGING_APP:
-				logEvent(this, "onActivityResult() :: returned from SMS app settings.  resultCode=%s, data=%s", resultCode, data);
 				// we should now know if we're the default SMS app from the value of
 				// resultCode, but it seems a little odd to trust that result when we
 				// can just check a method.
 				if(app.isDefaultSmsProvider(this)) {
 					continueToSettings();
 				}
 				break;
-			default: trace(this, "onActivityResult() :: No handling for requestCode: %s", requestCode);
+			default:
+				log("PromptToSetAsDefaultMessageAppActivity :: onActivityResult() :: No handling for requestCode: %s", requestCode);
 		}
 	}
 
 //> CUSTOM EVENT HANDLERS
-	public void dismissActivity(View v) {
+
+	public void dismissActivity(View view) {
 		continueToSettings();
 	}
 
-	public void openDefaultMessageAppSettings(View v) {
-		Intent i = new Intent(Intents.ACTION_CHANGE_DEFAULT);
-		i.putExtra(Intents.EXTRA_PACKAGE_NAME, getPackageName());
-		startActivityForResult(i, REQUEST_CHANGE_DEFAULT_MESSAGING_APP);
+	@TargetApi(Build.VERSION_CODES.KITKAT)
+	@SuppressLint({"ObsoleteSdkInt", "InlinedApi"})
+	public void openDefaultMessageAppSettings(View view) {
```

Comment:
I tried to unit test this change, but I couldn't find a nice way of mocking the Build.VERSION.SDK_INT or a way to test which intent got started by the activity, considering our current libs versions. 
Espresso lib didn't have IntentsTestRule or intents.init() 
Please let me know if you know a way to test this :)

Reply:
@ngaruko is adding better testing capabilities and coverage to the medic-android app in https://github.com/medic/medic-android/pull/173, so maybe you can talk with him about it.

Reply:
@ngaruko If you have any suggestion pls let me know :) For now moving this to AT.